### PR TITLE
[ITBL-5309] Add handleUniversalLink: method

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: Iterable-iOS-SDK/IterableAPI.h [hunk 1]
```diff
@@ -513,7 +513,16 @@ typedef void (^OnFailureHandler)(NSString *reason, NSData *_Nullable data);
  
  @discussion            passes the string of the redirected URL to the callback, returns the original webpageURL if not an iterable link
  */
-+(void) getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock;
++ (void)getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock;
```
Comment (tapashmajumder) [inline Iterable-iOS-SDK/IterableAPI.h:516]: Do we want to deprecate this call?

**Review by tapashmajumder — APPROVED** [review]
+1

File: Iterable-iOS-SDK/IterableAPI.m [hunk 2]
```diff
@@ -457,12 +457,19 @@ + (IterableAPI *)sharedInstanceWithApiKey:(NSString *)apiKey andUserId:(NSString
...
 
 // documented in IterableAPI.h
-+(void) getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock
++ (void)getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock
 {
     IterableDeeplinkManager *deeplinkManager = [IterableDeeplinkManager instance];
     [deeplinkManager getAndTrackDeeplink:webpageURL callbackBlock:callbackBlock];
 }
 
++ (void)resolveApplinkURL:(NSURL *)url callback:(ITBURLCallback)callback;
```
Comment (davidtruong) [inline Iterable-iOS-SDK/IterableAPI.m]: resolveAppLink sounds like a function that should automatically handle calling openUrl. 
Do you think we should do that automatically?

We can overload getAndTrackDeeplink to pass back a URL.

**Review by davidtruong — COMMENTED** [review]
Should we automatically handle the redirects and open the url?

File: Iterable-iOS-SDK/IterableAPI.m (hunk 2 above)
Comment (tapashmajumder) [reply Iterable-iOS-SDK/IterableAPI.m]: I think we should let the app handle logic for handling the url opens. This is how we have it in the sample app
```
        IterableAPI.resolve(applinkURL: url) { (resolvedUrl) in
            if let resolvedUrl = resolvedUrl {
                DeeplinkHandler.handle(url: resolvedUrl) // DeeplinkHandler is their class.
            }
        }

```

Comment (andyrova) [reply Iterable-iOS-SDK/IterableAPI.m]: Making the example `DeeplinkHandler` code available either in support docs or a repository would be helpful for clients

File: Iterable-iOS-SDK/IterableAPI.h [hunk 3]
```diff
@@ -513,7 +513,18 @@ typedef void (^OnFailureHandler)(NSString *reason, NSData *_Nullable data);
...
+
+/**
+ * Handles a Universal Link
+ * For Iterable links, it will track the click and retrieve the original URL,
+ * pass it to `IterableURLDelegate` for handling
+ * If it's not an Iterable link, it just passes the same URL `IterableURLDelegate`
```
Comment (tapashmajumder) [inline Iterable-iOS-SDK/IterableAPI.h]: passes the same URL *to* `IterableURLDelegate`

**Review by tapashmajumder — APPROVED** [review]

**Review by davidtruong — APPROVED** [review]
LGTM

**Pull request merged**